cibyl/models/ci/build.py

```diff
@@ -13,23 +13,29 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 """
+import inspect
 
 from cibyl.cli.argument import Argument
 from cibyl.models.attribute import AttributeValue
 
 
 class Build:
     """General model for a job build """
+    API = {
+        'build_id': {
+            'attr_type': str,
+            'arguments': [Argument(name='--build-id', arg_type=str,
+                                   description="Build ID")]
+        },
+        'status': {
+            'attr_type': str,
+            'arguments': [Argument(name='--build-status', arg_type=str,
+                                   description="Build status")]
+        },
+    }
+
     def __init__(self, build_id: str, status: str = None):
-        id_argument = Argument(name='--build-id', arg_type=str,
-                               description="Build id")
-        self.build_id = AttributeValue(name="build_id", attr_type=str,
-                                       value=build_id, arguments=[id_argument])
-        status_argument = Argument(name='--build-status', arg_type=str,
-                                   description="Build status")
-        self.status = AttributeValue(name="status", attr_type=str,
-                                     value=status,
-                                     arguments=[status_argument])
+        super().__init__(frame=inspect.currentframe())
```

Comment:
Why is there a need for this 'inspect.currentframe()'? As far as I have seen, its part of a reflection API that is not guaranteed to be implemented, so it would be best to find another approach if possible.

Reply:
Yes, I have removed it.